### PR TITLE
Refactor(eos_designs): port_channel_interfaces python_module as per eos_cli_config_gen

### PR DESCRIPTION
**Pull request by ghost**
## Change Summary

Refactor port_channel_interfaces python_module as per eos_cli_config_gen

### CODE REVIEW

File: ansible_collections/arista/avd/roles/eos_designs/python_modules/connected_endpoints/port_channel_interfaces.py [hunk 1]
```diff
@@ -33,7 +33,9 @@ def port_channel_interfaces(self) -> dict | None:
                 channel_group_id = get(adapter, "port_channel.channel_id", default=default_channel_group_id)
 
                 port_channel_interface_name = f"Port-Channel{channel_group_id}"
-                port_channel_interfaces[port_channel_interface_name] = self._get_port_channel_interface_cfg(adapter, channel_group_id, connected_endpoint)
+                port_channel_interfaces.append(
+                    {"name": port_channel_interface_name, **self._get_port_channel_interface_cfg(adapter, channel_group_id, connected_endpoint)}
```
Comment (gmuloc) [inline ansible_collections/arista/avd/roles/eos_designs/python_modules/connected_endpoints/port_channel_interfaces.py]: Can you replace this with having `self._get_port_channel_interface_cfg(adapter, channel_group_id, connected_endpoint)` taking an extra parameter `name`:  `self._get_port_channel_interface_cfg(adapter, port_channel_name, channel_group_id, connected_endpoint)`

and returning a dictionary with already `name` inside ?

File: ansible_collections/arista/avd/roles/eos_designs/python_modules/core_interfaces/port_channel_interfaces.py [hunk 2]
```diff
@@ -28,7 +28,7 @@ def port_channel_interfaces(self) -> dict | None:
             port_channel_interface = {key: value for key, value in port_channel_interface.items() if value is not None}
 
             interface_name = p2p_link["data"]["interface"]
-            port_channel_interfaces[interface_name] = port_channel_interface
+            port_channel_interfaces.append({"name": interface_name, **port_channel_interface})
```
Comment (gmuloc) [inline ansible_collections/arista/avd/roles/eos_designs/python_modules/core_interfaces/port_channel_interfaces.py]: same comment - can we investigate having `self._get_common_interface_cfg` returning the `name` as it is part of the `p2p_link["data"]` dictionary already ?

File: ansible_collections/arista/avd/roles/eos_designs/python_modules/network_services/port_channel_interfaces.py [hunk 3]
```diff
@@ -63,27 +64,31 @@ def port_channel_interfaces(self) -> dict | None:
...
                             key = f"{interface_name}.{subif['number']}"
-                            port_channel_interfaces[key] = {
-                                "type": "l2dot1q",
-                                "encapsulation_vlan": {
-                                    "client": {
-                                        "dot1q": {
-                                            "vlan": subif["number"],
+                            port_channel_interfaces.append(
+                                {
+                                    "name": key,
```
Comment (gmuloc) [inline ansible_collections/arista/avd/roles/eos_designs/python_modules/network_services/port_channel_interfaces.py]: let's take the opportunity to rename `key` in `subif_name` or something like this for clarity :)

File: ansible_collections/arista/avd/roles/eos_designs/python_modules/network_services/port_channel_interfaces.py [hunk 4]
```diff
@@ -100,11 +105,12 @@ def port_channel_interfaces(self) -> dict | None:
...
-                        port_channel_interfaces[interface_name] = interface
 
-            subif_parent_interfaces = {key: value for key, value in subif_parent_interfaces.items() if key not in port_channel_interfaces}
-            if subif_parent_interfaces:
-                port_channel_interfaces.update(subif_parent_interfaces)
+                        port_channel_interfaces.append(interface)
+
+            port_channel_interfaces.extend(
+                [subif_parent_interface for subif_parent_interface in subif_parent_interfaces if subif_parent_interface not in port_channel_interfaces]
+            )
```
Comment (gmuloc) [inline ansible_collections/arista/avd/roles/eos_designs/python_modules/network_services/port_channel_interfaces.py]: do you need the `[]` or does this work:
```suggestion
            port_channel_interfaces.extend(
                subif_parent_interface for subif_parent_interface in subif_parent_interfaces if subif_parent_interface not in port_channel_interfaces
            )
```


looks like it works fine:
```
>> test = []
>>> test.extend(i for i in range(1,10))
>>> test
[1, 2, 3, 4, 5, 6, 7, 8, 9]
>>>
```

File: ansible_collections/arista/avd/roles/eos_designs/python_modules/connected_endpoints/port_channel_interfaces.py (hunk 1 above)
Comment (Shivani-chourasiya) [reply ansible_collections/arista/avd/roles/eos_designs/python_modules/connected_endpoints/port_channel_interfaces.py]: Fixed

File: ansible_collections/arista/avd/roles/eos_designs/python_modules/network_services/port_channel_interfaces.py (hunk 3 above)
Comment (Shivani-chourasiya) [reply ansible_collections/arista/avd/roles/eos_designs/python_modules/network_services/port_channel_interfaces.py]: updated

**Review by gmuloc — APPROVED** [review]
LGTM

Couple of suprising-looking changes like bool -> str:
https://github.com/aristanetworks/ansible-avd/pull/2609/files#diff-927732ae69bc171772f5c00b4d41fc196006a148ee6fe38e131335a8f2a09379R542

and int -> str

https://github.com/aristanetworks/ansible-avd/pull/2609/files#diff-aa875e92b862e750be9b43c837b2d35809c7e314f4982c74067c8e8d479163b3R780

But in both cases it matches our schema so that good.

**Review by ClausHolbechArista — APPROVED** [review]

**Pull request merged**